src/common/middlewares/debug-logger.js

```
@@ -0,0 +1,59 @@
+    info: (message, data) => {
+      if (process.env.DEBUG_MODE !== 'true') return;
+      const timestamp = new Date().toISOString();
+      const logMessage = `[${timestamp}] [INFO] [${context}]: ${message} ${data ? JSON.stringify(data) : ''}`;
+      console.info("\x1b[32m%s\x1b[0m", logMessage);
+      debugLogStream.write(logMessage + '\n');
+    },
+    error: (message, data) => {
+      const timestamp = new Date().toISOString();
+      const logMessage = `[${timestamp}] [ERROR] [${context}]: ${message} ${data ? JSON.stringify(data) : ''}`;
+      console.error("\x1b[31m%s\x1b[0m", logMessage);
+      debugLogStream.write(logMessage + '\n');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)